### PR TITLE
MM-23120 Custom status issues

### PR DESCRIPTION
**Pull request by manojmalik20**
#### Summary

- Fixed MM-34095 Show recent statuses on the set status confirmation screen for more efficient status switching
- Fixed MM-33149 Updated "Mobile Push Notifications" text in Account Settings
- Fixed MM-33195  Custom Status emoji misaligned in profile popover
- Fixed MM-33246 Custom Status Modal: increase max height so that all default options are visible without scrolling
- Fixed MM-33433 Custom statuses should appear in channel switcher
- Fixed MM-34571 Tooltip for custom status without text has emoji misaligned
- Fixed MM-33431 Custom status emoji size match the emoji size used in reactions and alignment needs tweaking
- Fixed MM-33214 With RHS open, pin icon and channel description overlap statuses in lower browser widths
- Fixed MM-33220 Reduce opacity of custom status emoji in LHS DMs

#### Ticket Link

- https://mattermost.atlassian.net/browse/MM-34095
- https://mattermost.atlassian.net/browse/MM-33149
- https://mattermost.atlassian.net/browse/MM-33195
- https://mattermost.atlassian.net/browse/MM-33246
- https://mattermost.atlassian.net/browse/MM-33433
- https://mattermost.atlassian.net/browse/MM-34571
- https://mattermost.atlassian.net/browse/MM-33431
- https://mattermost.atlassian.net/browse/MM-33214
- https://mattermost.atlassian.net/browse/MM-33220

#### Related Pull Requests
https://github.com/mattermost/mattermost-webapp/pull/7750

### CODE REVIEW

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Thanks @manojosh! Nice work resolving these issues. I just have a few really minor points:

1. In the Channel Switcher, can we make the left margin on the custom status 8px instead of 4px? It feels like it needs a bit more space there
2. This may not be related to this PR, but the bullet after the status in the DM channel header feels odd now. I wonder if we should just consider removing this since the spacing may provide enough differentiation here
Screen Shot 2021-04-08 at 11 29 20 AM
3. For the reduced opacity emojis in the left sidebar, can we make them full opacity when you hover over them?

**Comment by esethna** [comment]
Thanks @manojosh, one thing I noticed: The modal height should dynamically resize to accommodate all selections as noted in the ticket (note that the max options showing would be 10: 5 recent and 5 suggestions), right now it is scrolling to reveal all options: https://mattermost.atlassian.net/browse/MM-33246
image

**Comment by manojosh** [comment]
@matthewbirtch All the changes are done.

@esethna If we don't want any scrolling to accommodate all the 10 suggestions then the modal looks like this.

Screenshot from 2021-04-09 13-36-32

So we have to change the positioning of the modal. Right now, it's done by giving margin-top to the modal as **calc(50vh - 240px)**
I thought of changing it to **calc(50vh - 320px)**. Here is how it looks

Screenshot from 2021-04-09 13-47-09
Screenshot from 2021-04-09 13-47-27

Is this fine or do you want it to be dynamically changed as the no. of recent statuses increase? If so, what is the minimum value of margin-top which I should use?

**Comment by matthewbirtch** [comment]
> @matthewbirtch All the changes are done.
> 
> @esethna If we don't want any scrolling to accommodate all the 10 suggestions then the modal looks like this.
> 
> Screenshot from 2021-04-09 13-36-32
> 
> So we have to change the positioning of the modal. Right now, it's done by giving margin-top to the modal as **calc(50vh - 240px)**
> I thought of changing it to **calc(50vh - 320px)**. Here is how it looks
> 
> Screenshot from 2021-04-09 13-47-09
> Screenshot from 2021-04-09 13-47-27
> 
> Is this fine or do you want it to be dynamically changed as the no. of recent statuses increase? If so, what is the minimum value of margin-top which I should use?

@esethna I think we should consider this effort as a separate PR if we want to merge this in for the next release. Accommadating for 10 options without scrolling makes for a pretty tall modal and I don't think we should do this. Also, it wouldn't work well for people on smaller resolution. Can we leave it as-is for now and create a follow-up ticket to look in to dynamically resizing if we feel strongly?

**Comment by esethna** [comment]
@matthewbirtch @manojosh okay that's fine, we can leave as is. But I'm not clear on what changed in this PR from the existing behaviour? Can we just update the ticket to reflect what changed so QA can test appropriately?

**Comment by manojosh** [comment]
@esethna Actually what changed was the min and max height of the modal. Earlier if any one of the suggestions was present in the recents then the **Recents** title would appear along with the scrollbar but now the height is adjusted to accommodate all the suggestions along with the **Recents** and **Suggestions** titles.

Refer to the gif here https://mattermost.atlassian.net/browse/MM-33246

**Comment by matthewbirtch** [comment]
> 1. In the Channel Switcher, can we make the left margin on the custom status 8px instead of 4px? It feels like it needs a bit more space there
> 2. This may not be related to this PR, but the bullet after the status in the DM channel header feels odd now. I wonder if we should just consider removing this since the spacing may provide enough differentiation here
> 
> Screen Shot 2021-04-08 at 11 29 20 AM
> 
> 3. For the reduced opacity emojis in the left sidebar, can we make them full opacity when you hover over them?

These changes all look to be resolved now. I'll approve

**Review by matthewbirtch — APPROVED** [review]

**Comment by esethna** [comment]
> height is adjusted to accommodate all the suggestions along with the **Recents** and **Suggestions** titles.

Got it, I adjusted the ticket description, thanks

**Comment by esethna** [comment]
@michelengelen @Willyfrog for dev review, thanks!

File: components/custom_status/custom_status_modal.tsx [hunk 1]
```diff
@@ -189,6 +189,10 @@ const CustomStatusModal: React.FC<Props> = (props: Props) => {
         </div>
     );
 
+    const showSuggestions = (currentCustomStatus && currentCustomStatus.emoji) ?
+        (!isStatusSet || (currentCustomStatus.emoji === emoji && text && currentCustomStatus.text === text)) :
+        !isStatusSet;
+
```
Comment (michelengelen) [inline components/custom_status/custom_status_modal.tsx:195]: couldn't this be simpified by using nullish coalescent?

```suggestion
    const showSuggestions = !isStatusSet || (currentCustomStatus?.emoji === emoji && text && currentCustomStatus?.text === text);
        
```

**Review by michelengelen — APPROVED** [review]
Only a small suggestion from my side ... otherwise LGTM! Thanks for your contribution! 🙇🏻‍♂️

**Review by Willyfrog — APPROVED** [review]
LGTM

**Comment by esethna** [comment]
Friendly ping @ogi-m for QA review.

**Comment by ogi-m** [comment]
#### Verified the following:
- MM-34095
- MM-33149
- MM-33195
- MM-33433
- MM-34571
- MM-33431
- MM-33214
- MM-33220

#### Two questions:
1. The emoji seem to me like they're sitting too low now in the LHS header? @matthewbirtch
It's more apparent with squared emoji and not with faces.
PR vs master:
PR daily


2. If I'm understanding correctly, [the ticket for showing all options](https://mattermost.atlassian.net/browse/MM-33246) hasn't been implemented in the PR and will be done separately? @esethna

**Comment by matthewbirtch** [comment]
> 1. The emoji seem to me like they're sitting too low now in the LHS header? @matthewbirtch
>    It's more apparent with squared emoji and not with faces.
>    PR vs master:

It's weird the calendar emoji specifically seems to sit a bit low. Most other emojis seems to sit appropriately. I think it's fine.
image

**Review by michelengelen — APPROVED** [review]

**Comment by esethna** [comment]
@ogi-m re # 2 ^, correct, it is not part of this PR

**Review by ogi-m — APPROVED** [review]
Thanks guys, all good then! Approving 👍🏻

**Comment by amyblais** [comment]
@esethna Sure, why not in Cloud April 21st release?

**Comment by esethna** [comment]
@amyblais sure if there's still time for that release please go ahead

**Comment by michelengelen** [comment]
Hi @manojosh ... I am sorry, but current changes to master created conflicts with your branch. Could you please merge master again and resolve the conflicts? I will merge it as soon as changes are coming in! 

Again ... I am sorry this causes trouble for you! 🙇🏻‍♂️

**Comment by manojosh** [comment]
No problem @michelengelen.. Resolving the conflicts

**Comment by michelengelen** [comment]
Thanks a lot @manojmalik20 ... will merge it now!

**Comment by amyblais** [comment]
@Willyfrog Seems that the cherry-pick didn't work

**Comment by Willyfrog** [comment]
yeah! I'll try to do that manually

**Comment by Willyfrog** [comment]
actually @michelengelen can you do the cherry-picking? seems the reason for the conflict is that https://github.com/mattermost/mattermost-webapp/pull/7932 is not in both branches

**Pull request merged**